rust/agama-lib/src/hostname/client.rs

```diff
@@ -51,16 +51,23 @@ impl<'a> HostnameClient<'a> {
 
     pub async fn set_config(&self, config: &HostnameSettings) -> Result<(), ServiceError> {
         let settings = self.get_config().await?;
-        if settings.hostname != config.hostname {
-            self.hostname_proxy
-                .set_hostname(config.hostname.as_str(), false)
-                .await?;
+
+        // order is important as otherwise the transient hostname could not be set in case the
+        // static one is not empty
+        if let Some(config_static_hostname) = config.static_hostname.clone() {
```

Comment:
this clone looks quite strange in let condition. maybe it should be cloned later if needed?